integration/tests/basic/evm/test_solana_interoperability.py

```diff
@@ -303,3 +412,251 @@ def test_limit_of_simple_instr_in_one_trx(self, call_solana_caller, counter_reso
+            sol_client.send_sol(bank_account, from_wallet.pubkey(), int(0.5 * 10**9))
+        else:
+            sol_client.request_airdrop(from_wallet.pubkey(), 1000 * 10**9, commitment=Confirmed)
+
+        serialized, _, _ = self.get_transfer_instruction(sol_client, from_wallet, to_wallet, amount, call_solana_caller, False)
+
+        tx = self.web3_client.make_raw_tx(from_=sender.address, estimate_gas=True)
+
+        instruction_tx = call_solana_caller.functions.executeInIterativeMode(iterations, 0, serialized).build_transaction(tx)
+
+        resp = self.web3_client.send_transaction(sender, instruction_tx)
+        assert resp["status"] == 0
+
+        event_logs = call_solana_caller.events.LogStr().process_receipt(resp)
+        assert len(event_logs) == 0
+
+
+    def test_solana_call_after_iterative_actions_exceed_accounts_limit(self, counter_resource_address, call_solana_caller):
+        iterations = 53
+        sender = self.accounts[0]
+        lamports = 0
+
+        instruction = Instruction(
+            program_id=COUNTER_ID,
+            accounts=[
+                AccountMeta(Pubkey(counter_resource_address), is_signer=False, is_writable=True),
+            ],
+            data=bytes([0x1]),
+        )
+        serialized = serialize_instruction(COUNTER_ID, instruction)
+
+        tx = self.web3_client.make_raw_tx(sender.address)
+
+        with pytest.raises(
+            web3.exceptions.ContractLogicError,
+            match="too many accounts: 65 > 64",
+        ):
+            call_solana_caller.functions.executeInIterativeMode(iterations, lamports, serialized).build_transaction(tx)
+
+    def test_solana_call_before_iterative_actions(self, counter_resource_address, call_solana_caller, get_counter_value):
+        iterations = 29
+        sender = self.accounts[0]
+        lamports = 0
+
+        instruction = Instruction(
+            program_id=COUNTER_ID,
+            accounts=[
+                AccountMeta(Pubkey(counter_resource_address), is_signer=False, is_writable=True),
+            ],
+            data=bytes([0x1]),
+        )
+        serialized = serialize_instruction(COUNTER_ID, instruction)
+
+        tx = self.web3_client.make_raw_tx(sender.address)
+        instruction_tx = call_solana_caller.functions.executeAndDoSomeIterativeActions(iterations,
+                                                                                       lamports,
+                                                                                       serialized).build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender, instruction_tx)
+        assert resp["status"] == 1
+        event_logs = call_solana_caller.events.LogBytes().process_receipt(resp)
+        assert int.from_bytes(event_logs[0].args.value, byteorder="little") == next(get_counter_value)
+
+    def test_iterative_actions_after_solana_call_exceed_accounts_limit(self, counter_resource_address, call_solana_caller):
```

Comment:
I think  this test can be deleted as previous one is negative

Reply:
I want to keep it cause the reason here is account limit, previous test is positive, the third one is negative but with evm memory limit